### PR TITLE
Feature/validations for application draft

### PR DESCRIPTION
**Pull request by ramonh**
Together with @carpodaster, I've added validations for the `Student` fields, along with that of those of the `Coach`, and the `ApplicationDraft` itself.

### CODE REVIEW

**Comment by carpodaster** [comment]
Hm, it doesn't have all the patches. Can you PR against `rails-girls-summer-of-code:master` instead of `rails-girls-summer-of-code:feature/validations-for-application-draft`? That probably doesn't work after the PR has been issued and you may have to close this PR again.

screen shot 2015-03-17 at 20 46 13

**Comment by carpodaster** [comment]
Then again, nevermind. We know the missing bits were working on sunday and this patchset is less noisy than comparing the whole set.

**Comment by ramonh** [comment]
Oops, not sure I understood. I forked off the branch we were working on on Sunday

File: spec/models/application_draft_spec.rb [hunk 1]
```diff
@@ -37,17 +57,48 @@
...
+          end
         end
 
         context 'both being valid' do
-          skip
+          let(:ace_student_2)   { double.as_null_object }
+          let(:value_2)         { SecureRandom.hex(12)  }
+
+          before do
+            allow(subject).to receive(:students).and_return([ace_student, ace_student_2])
```
Comment (carpodaster) [inline spec/models/application_draft_spec.rb]: You should be able to re-use the `ace_student` from up above as we don't have a mechanism in place that would check for the students not being the same reference. If it works w/o, lines 88-90 can be removed as well.

File: spec/models/application_draft_spec.rb [hunk 2]
```diff
@@ -6,6 +6,26 @@
   context 'with validations' do
     it { is_expected.to validate_presence_of :team }
 
+    context 'apply validations' do
+      let(:student0) { Student.new }
+
+      before do
+        allow(subject).to receive(:students).and_return([student0])
```
Comment (carpodaster) [inline spec/models/application_draft_spec.rb]: Will it break if we leave out stubbing the `students` tuple?

**Comment by ramonh** [comment]
All done! Thanks for the feedback.

**Comment by ramonh** [comment]
Yay!

**Comment by sareg0** [comment]
Fantastic your guys!!

**Pull request merged**